src/lib/reasoners/fun_sat.ml

```diff
@@ -1753,6 +1755,10 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
       (* don't attempt to compute a model if timeout before
          calling unsat function *)
       i_dont_know env (Timeout Assume)
+    | Util.Step_limit_reached _ ->
+      (* When reaching the step limit on an assume, we do not want to answer
+         'unknown' right away. *)
+      env
```

Comment:
I believe that this is still leaking a potentially environment into the wild without *protecting* it with an unknown reason. This should raise `i_dont_know` and it should be caught in `handle_sat_exn` in the frontend to set the state there to `` `Unknown`` and following operations will be protected by `wrap_f`.

Reply:
Sorry, this comment shouldn't be here — it should be in the corresponding position in `satml_frontend`